src/ReachSets/ContinuousPost/GLGM06/reach.jl

```diff
@@ -0,0 +1,76 @@
+# ===============================================================
+# Homogeneous case
+# ===============================================================
+function reach_homog!(HR::Vector{ReachSet{Zonotope, Float64}},
+                      Ω0::Zonotope,
+                      Φ::AbstractMatrix,
+                      N::Int,
+                      δ::Float64,
+                      max_order::Int)
+
+    # save timestamps with the reach set
+    t0, t1 = zero(δ), δ
+
+    # initial reach set
+    HR[1] = ReachSet{Zonotope, Float64}(Ω0, t0, t1)
+
+    k = 1
+    while k < N
+        HR_next = linear_map(Φ, HR[k].X)
+        if order(HR_next) > max_order
+            HR_next = reduce_order(HR_next, max_order)
+        end
+        t0 = t1; t1 += δ
+        HR[k+1] = ReachSet{Zonotope, Float64}(HR_next, t0, t1)
+        k = k + 1
+    end
+    return HR
+end
+
+# ===============================================================
+# Inhomogeneous case
+# ===============================================================
+function reach_inhomog!(X::Vector{ReachSet{Zonotope, Float64}},
+                        Ω0::Zonotope,
+                        U::ConstantInput,
+                        Φ::AbstractMatrix,
+                        N::Int,
+                        δ::Float64,
+                        max_order::Int)
+
+    # preallocation
+    W = Vector{Zonotope}(undef, N+1)
+
+    # initialization
+    k = 1
+    n = size(Φ, 1)
+    t0, t1 = zero(δ), δ
+    X[k] = ReachSet{Zonotope, Float64}(Ω0, t0, t1)
+    W[k] = ZeroSet(n)
```

Comment:
Why does this not crash (`W` only stores `Zonotope`s)? Actually, you never access `W[1]`. Is this correct?

Reply:
Another comment: It seems that you do not need to store all the inputs. You always use the current index, so it suffices to store a single set.

Reply:
> Why does this not crash 

Julia does the conversion automatically:

```julia
julia>  W = Vector{Zonotope}(undef, 2)
2-element Array{Zonotope,1}:
 #undef
 #undef

julia> W[1] = ZeroSet(2)
ZeroSet{Float64}(2)

julia> W
2-element Array{Zonotope,1}:
    Zonotope{Float64}([0.0, 0.0], [0.0 0.0; 0.0 0.0])
 #undef
```

> it suffices to store a single set.

Good point!